comp/autoscaling/datadogclient/mock/mock.go

```diff
@@ -0,0 +1,61 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build test
+
+// Package mock provides mock methods
+package mock
+
+import (
+	"sync"
+	"testing"
+
+	datadogclient "github.com/DataDog/datadog-agent/comp/autoscaling/datadogclient/def"
+	"gopkg.in/zorkian/go-datadog-api.v2"
+)
+
+type mockDatadogClient struct {
+	mux               sync.RWMutex
+	queryMetricsFunc  func(from, to int64, query string) ([]datadog.Series, error)
+	getRateLimitsFunc func() map[string]datadog.RateLimit
+}
```

Comment:
What do you think about using https://pkg.go.dev/github.com/stretchr/testify@v1.9.0/mock instead of writing a custom mock ?